File: code/java/interview_arena_article-03.java
public class Interview_arena_article_03 {
    /**
     * Algorithm problems that can be solved with one line of code
     */
    public Object solve(Object... args) {
        if (args == null || args.length != 1) {
            throw new IllegalArgumentException("Expected: (int n)");
        }

        Solution solution = new Solution();
        return solution.canWinNim((int) args[0]);
    }

    private static class Solution {

        public boolean canWinNim(int n) {
            return n % 4 != 0;
        }

    }

}

File: code/python/interview_arena_article-03.py
"""Algorithm problems that can be solved with one line of code"""

from __future__ import annotations
from typing import Any


def solve(*args: Any, **kwargs: Any) -> Any:
    """Run this lesson exercise with positional inputs."""
    return _Solution().canWinNim(*args, **kwargs)


class _Solution:
    def canWinNim(self, n: int) -> bool:
        return n % 4 != 0
